content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+
+if p <= alpha:
+ print('Reject the Null Hypothesis ( reject H0)')
+else: 
+ print('Accept the Null Hypothesis (fail to reject H0)')
+```
+![alphsig](alphasiglevel.png)
+
+The P-value is greater than the alpha value set at 0.05, we will fail to reject the null hypothesis(we accept the null hypothesis). This means the petal width and species of flower are not dependent, i.e. there is no relationship between them.
+
+#### T-Test
+
+The T-test is utilized when we plan to evaluate the discrepancy between the means of two groups of samples. Unlike Chi-Square Test, the T-test is used on continuous variables.
+
+It has three basic types
+
+1. **One sample t-test:** It checks whether the population has a different mean from the sample.
+The mathematical formula is:
+```
+t-test = a-BX∕N
+```
+Where;
+
+a = the observed mean group.
+
+B = the assumed population means.
+
+X = the standard deviation of the data group.
+
+N = the number of observations in the group.
+
+2. **Two-sample t-test:** It checks and compares the means of two groups that are not dependent on each other and compare with the population's means to see if there is a huge difference.
+The mathematical expression is:
+```
+t-test = Ma – (Mb*v*Na)+vNb
+```
+Ma = Data sample A's mean
+
+Mb = Data sample B's mean
+
+Na = Size of sample A
+
+Nb = Size of sample B
+
+v = Variance
+
+3. **Paired t-test:** It is used in checking and comparing the means of different samples from a group.
+Mathematically, it can be resolved with:
+```
+t-test = D - (m*xd)∕N
+```
+D = The difference between paired observations' sample mean.
+
+m = assumed mean difference.
+
+xd = standard deviation.
+
+N = the number of observations in the group.
+
+#### Performing a One-Sample T-Test
+```python
+df_ = df.petal_width
+```
+```python
+df_.sample(5)
+```
+![df_sample](df_sample5.png)
+```python
+stats.ttest_1samp(a=df_, popmean=1.199)
+```
+![onesampttest](onsamplettest.png)
+
+**H0:** The mean of petal_width is 1.199
+
+**H1:** The mean of petal_width is not 1.199
+
+Analyzing one-sample t-test, it finds out if the hypothesized mean is the same or different from the group's mean. From the example above, we've selected the petal_width as the population sample to perform the test on. 
+From the test, since the P-value beats the alpha level set at 0.05,
+
+we fail to reject the null hypothesis because we don't have enough evidence to prove otherwise, moreover, if you check, the P-value is very close to 1.0,
+
+This means that the mean of petal_width is most likely 1.199.
+ 
+ The evidence is too strong to be rejected, it is almost accurate. (Obviously, I copied the mean from the summary).
+
+#### Performing a Two-Sample T-Test
+
+We need to first of all check their variances.
+
+The ratio of the higher to the lower variance must be less than 4:1 before we can assume the two samples have the same variance.
+```python
+class1 = df.petal_width
+class2 = df.sepal_length
+```
+find variance for each group
```

Comment:
This sentence is not punctuated at all.
I can see that it was intended to be a Python comment, so do make the appropriate change.